### DISCUSSION TITLE
YOLOv8m is fast on iPhone 13 mini but slow on Macbook M1 Pro

### DISCUSSION THREAD

**Discussion by olehmisar** [ultralytics/ultralytics#4365] [Q&A] [answered]
I just downloaded Ultralytics HUB on my phone and it runs YOLOv8m at 24 FPS on average. Then I installed ultralytics on my laptop and ran a prediction of a 10-second video ([link](https://drive.google.com/file/d/115wx-z8SVKL4VEN8WpqTXkYfjUOGSYHO/view?usp=drive_link)) using 
```sh
yolo predict model=yolov8m.pt source='./test1.mp4'
```
It took a couple of minutes to finish the prediction. At the end it showed these stats:
```
Speed: 2.1ms preprocess, 303.1ms inference, 0.9ms postprocess per image at shape (1, 3, 384, 640)
```

303ms for inference is roughly equal to 3 FPS. Why is it that slow on my laptop while on my phone it is running in real time? Am I missing something?

Hardware specs:
- iPhone 13 mini
- Macbook M1 Pro 16/512

**Comment by olehmisar** [comment, marked as answer]
Adding `device=mps` improves the speed on my M1 cpu **SIGNIFICANTLY**. Now, it's 23.9ms inference vs 303.1ms inference.

```sh
yolo predict model=yolov5nu.pt source='./test1.mp4' device=mps
```

```
Speed: 3.0ms preprocess, 23.9ms inference, 11.6ms postprocess per image at shape (1, 3, 384, 640)
```

Found this trick in [this article](https://pysource.com/2023/03/28/object-detection-with-yolo-v8-on-mac-m1/)

**Reply by glenn-jocher** [reply]
@olehmisar great to hear that specifying `device=mps` has given you a significant speed boost on your Macbook M1 Pro! The Metal Performance Shaders (MPS) backend is optimized for Apple silicon and can indeed provide much faster inference times compared to the default CPU processing. It's always satisfying when a simple change yields such a noticeable performance improvement. Keep exploring and optimizing, and if you have any more questions or need further assistance, feel free to reach out. Happy detecting with YOLOv8! 🚀😊
